src/components/utils.ts

```diff
@@ -71,6 +71,13 @@ export const getAlertsAndRules = (
     return _.filter(g.rules, { type: 'alerting' }).map(addID);
   });
 
+  // Add external labels to all `rules[].alerts[].labels`
+  rules.map((rule) => {
+    if (rule.alerts.length > 0) {
+      rule.alerts.map((alert) => (alert.labels = { ...rule.labels, ...alert.labels }));
+    }
+  });
```

Comment:
It seems `map` is not needed here:

```suggestion
  rules.forEach((rule) => {
    if (rule.alerts.length > 0) {
      rule.alerts.forEach((alert) => (alert.labels = { ...rule.labels, ...alert.labels }));
    }
  });
```

Reply:
Thanks! Committed and squashed in the latest commit.